zio-http/src/main/scala/zhttp/http/Cookie.scala

```diff
@@ -176,24 +246,24 @@ object Cookie {
     val cookies: Array[String]  = headerValue.split(";").map(_.trim)
     val x: List[Option[Cookie]] = cookies.toList.map(a => {
       val (name, content) = splitNameContent(a)
-      if (name.trim == "" && content.isEmpty) None
-      else Some(Cookie(name, content.getOrElse("")))
+      if (name == "" && content.isEmpty) None
+      else Some(Cookie(name, content))
     })
 
     if (x.contains(None))
       None
     else Some(x.map(_.get))
   }
 
-  private def parseDate(v: String): Option[Instant] =
-    Try(Instant.parse(v)).toOption
-
-  private def splitNameContent(kv: String): (String, Option[String]) =
-    kv.split("=", 2).map(_.trim) match {
-      case Array(v1)     => (v1, None)
-      case Array(v1, v2) => (v1, Some(v2))
-      case _             => ("", None)
+  @inline
+  private def splitNameContent(str: String): (String, String) = {
```

Comment:
Avoid tuple creation.

Reply:
You can create two separate functions that give you different parts from the string.

Reply:
splitNameContent is currently only used in decodeRequestCookie, to avoid this tuple allocation we can just chante it to
```scala
val name    =
  if (i >= 0) a.substring(0, i).trim
  else a.trim
val content =
  if (i >= 0) a.substring(i + 1).trim
  else null

if (name.isEmpty && content.isEmpty) None
else Some(Cookie(name, content))
```

however, maybe this is the best candidate to use a index base view, like  you mentioned earlier ...
```
case class CookieView(cookieStr) implements Cookie {
  def name: String = ...
  def content: String = ....
}
```